### PR TITLE
feat(router): add offset in mandate list route

### LINKED ISSUE #3748: [FEATURE]: add `offset` field to mandates list
### Feature Description

The [mandates list](https://github.com/juspay/hyperswitch/blob/54c0b218d9f20c6388dde0f30ea525011cdad573/crates/router/src/routes/mandates.rs#L114) endpoint currently has no field to specify `offset`, which would skip certain number of entries from the start. This is necessary in order to provide pagination support.

### Possible Implementation

accept a field called `offset` which specifies the number of items to skip from the start.

This would be similar to the `payments_list` pagination. Refer [here](https://github.com/juspay/hyperswitch/blob/54c0b218d9f20c6388dde0f30ea525011cdad573/crates/router/src/routes/payments.rs#L868)

### Have you spent some time checking if this feature request has been raised before?

- [X] I checked and didn't find a similar issue

### Have you read the Contributing Guidelines?

- [X] I have read the [Contributing Guidelines](https://github.com/juspay/hyperswitch/blob/main/docs/CONTRIBUTING.md)

### Are you willing to submit a PR?

None

### PR DESCRIPTION
**Pull request by PanGan21**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description

Add offset to mandate list endpoint.
Closes #3748 

### Additional Changes

- [x] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## How did you test it?

- Create 10 mandates.
- Use the below curl to list the first 5 mandates.
```bash
curl --location 'http://localhost:8080/mandates/list?limit=5' \
--header 'api-key: dev_dg0YfZam6V6VM2KBPFLr0olm6kGM3QojUE9Ba6f8Gzzi7IIAnkN4fyfKsyc55pB4'
```

- Use the below curl to list next 5 mandates.
```bash
curl --location 'http://localhost:8080/mandates/list?limit=5&offset=5' \
--header 'api-key: dev_dg0YfZam6V6VM2KBPFLr0olm6kGM3QojUE9Ba6f8Gzzi7IIAnkN4fyfKsyc55pB4'
```

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

**Comment by PanGan21** [comment]
Hi @Narayanbhat166 maybe you could have a look at this pr which is related to #3748  ?

**Review by Narayanbhat166 — DISMISSED** [review]

**Comment by Narayanbhat166** [comment]
Hey @PanGan21, The PR Looks Good to me.

**Review by apoorvdixit88 — DISMISSED** [review]

File: crates/router/src/types/storage/mandate.rs [hunk 1]
```diff
@@ -54,6 +54,9 @@ impl MandateDbExt for Mandate {
         if let Some(limit) = mandate_list_constraints.limit {
             filter = filter.limit(limit);
         }
+        if let Some(offset) = mandate_list_constraints.offset {
+            filter = filter.offset(offset);
+        }
```
Comment (SanchithHegde) [inline crates/router/src/types/storage/mandate.rs:59]: Could you make the corresponding change in the `MockDb` implementation of `MandateInterface` as well?

https://github.com/juspay/hyperswitch/blob/7391416e2473eab0474bd01bb155a9ecc96da263/crates/router/src/db/mandate.rs#L233-L275

**Review by SanchithHegde — COMMENTED** [review]
Other than that, looks good to me!

File: crates/router/src/types/storage/mandate.rs (hunk 1 above)
Comment (PanGan21) [reply crates/router/src/types/storage/mandate.rs:59]: It should be fixed now! :)

**Review by SanchithHegde — APPROVED** [review]

**Review by Narayanbhat166 — APPROVED** [review]

**Review by apoorvdixit88 — APPROVED** [review]

**Pull request merged**